components/Card.vue

```diff
@@ -230,6 +238,17 @@ export default Vue.extend({
         return ImageType.Primary;
       }
     }
+  },
+  created() {
+    this.$store.subscribe((mutation, state) => {
+      if (
+        mutation.type === 'SOCKET_ONMESSAGE' &&
+        state.socket.message.MessageType === 'RefreshProgress' &&
+        state.socket.message.Data.ItemId === this.item.Id
+      ) {
+        this.refreshProgress = state.socket.message.Data.Progress;
+      }
+    });
```

Comment:
I know this is only a draft, but it is probably better to check the card type before subscribing to the websocket messages

Reply:
The idea here is that other items could also get refresh progress down the line.

There are bits in the server about provider refresh progress which, while unused now, could be used in the future to inform the user about individual item refreshes.

However, for now, we could limit it to libraries.